### PR TITLE
Fix issues in GCP auto-discovery docs

### LINKED ISSUE #31382: GCP Discovery AWK command is not escaped correctly
## Applies To

https://goteleport.com/docs/server-access/guides/gcp-discovery/#enable-guest-attributes-on-instances

## Details

The documentation provides the following shell script:
```
#!/usr/bin/env bash
for file in /etc/ssh/ssh_host_*_key.pub
  do
  KEY_TYPE=$(awk '\''{print $1}'\'' $file)
  KEY=$(awk '\''{print $2}'\''  $file)
  curl -X PUT --data "$KEY" "http://metadata.google.internal/computeMetadata/v1/instance/guest-attributes/hostkeys/$KEY_TYPE" -H "Metadata-Flavor: Google"
done
```

When I run it, I get escape errors:
```
./add-host-keys.sh: line 5: unexpected EOF while looking for matching `''
./add-host-keys.sh: line 8: syntax error: unexpected end of file
```

## How will we know this is resolved?

The command will work in GCP Ubuntu shell.

When I remove the extra `'`, it seems to work:
```
cat add-host-keys.sh 
#!/usr/bin/env bash
for file in /etc/ssh/ssh_host_*_key.pub
  do
  KEY_TYPE=$(awk '{print $1}' $file)
  KEY=$(awk '{print $2}'  $file)
  curl -X PUT --data "$KEY" "http://metadata.google.internal/computeMetadata/v1/instance/guest-attributes/hostkeys/$KEY_TYPE" -H "Metadata-Flavor: Google"
done
```

### LINKED ISSUE #31388: GCP discovery config should say labels instead of tags
Right now the discovery config uses `tags` as a filter name:

```yaml
discovery_service:
  enabled: "yes"
  gcp:
    - types: ["gce"]
      project_ids: ["<redacted>"]
      locations: ["us-central1-a"]
      tags:
        "project": "*"
```

But in the code we actually match against labels: https://github.com/gravitational/teleport/blob/bf95126ea140aed72f07678fb2666b952afdcedb/lib/cloud/gcp/vm.go#L173

It is very confusing because users set instance tags and then wonder why matching doesn't work.

### LINKED ISSUE #31813: GCP join token mismatch between code and docs
The guide tells users to create token named `gcp-token`:

https://goteleport.com/docs/server-access/guides/gcp-discovery/

The code uses `gcp-discovery-token` as the default token name:

https://github.com/gravitational/teleport/blob/9c55a33e0143f1e8bd6063872e139705a2e47941/lib/defaults/defaults.go#L916

Should these match?

### PR DESCRIPTION
**Pull request by atburke**
This PR fixes multiple issues in the GCP discovery docs:
- Fix mismatch between GCP token names. Resolves #31813.
- Fix awk quote escaping in sample script. Resolves #31382.
- Rename `tags` to `labels` in matchers and config to reflect that we match to GCP labels rather than GCP tags. `tags` is still available for backwards compatibility. Resolves #31388.

### CODE REVIEW

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -6197,7 +6197,7 @@ message GCPMatcher {
   repeated string Types = 1 [(gogoproto.jsontag) = "types,omitempty"];
   // Locations are GKE locations to search resources for.
   repeated string Locations = 2 [(gogoproto.jsontag) = "locations,omitempty"];
-  // Tags are GCP labels to match.
+  // Tags are an alias for Labels.
```
Comment (zmb3) [inline api/proto/teleport/legacy/types/types.proto:6200]: I'm not sure what this means. Are you supposed to configure tags, labels, or both? What happens if they differ? Why do we need two fields to begin with?

Comment (r0mant) [inline api/proto/teleport/legacy/types/types.proto:6200]: We should mark this field as obsolete (even with just a comment).

Comment (r0mant) [reply api/proto/teleport/legacy/types/types.proto:6200]: @zmb3 This is a result of poor naming choice on our part. GCP has a concept of both "tags" and "labels". Our auto-discovery looks at "labels" but the config field was called "tags" so it was very confusing. Now we're basically deprecating "tags" in our config and replacing with "labels" but we can't remove it completely for compatibility.

@atburke Let's make it clear that this field is obsolete, not supposed to be used by users and only exists for backwards compatibility.

File: api/types/matchers.go [hunk 2]
```diff
@@ -59,3 +59,11 @@ func (m GCPMatcher) CopyWithTypes(t []string) Matcher {
 	newMatcher.Types = t
 	return newMatcher
 }
+
+// GetLabels gets the matcher's labels.
+func (m GCPMatcher) GetLabels() Labels {
+	if len(m.Labels) != 0 {
+		return m.Labels
+	}
+	return m.Tags
```
Comment (r0mant) [inline api/types/matchers.go:68]: Add a comment about backwards compatibility here as well.

File: lib/srv/discovery/discovery_test.go [hunk 3]
```diff
@@ -827,7 +827,7 @@ func TestDiscoveryInCloudKube(t *testing.T) {
 					Types:      []string{"gke"},
 					Locations:  []string{"*"},
 					ProjectIDs: []string{"p1"},
-					Tags:       map[string]utils.Strings{"env": {"prod"}},
+					Labels:     map[string]utils.Strings{"env": {"prod"}},
```
Comment (r0mant) [inline lib/srv/discovery/discovery_test.go:830]: We need to keep one of the tests that uses Tags to test backwards compatibility.

**Review by r0mant — APPROVED** [review]

**Review by probakowski — APPROVED** [review]

**Pull request merged**